llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -2823,6 +2857,9 @@ bool RISCVTTIImpl::isProfitableToSinkOperands(
   if (!ST->sinkSplatOperands())
     return false;
 
+  if (isa<VPIntrinsic>(I) && tryToSinkVPSplat(cast<VPIntrinsic>(I), Ops))
+    return true;
+
   for (auto OpIdx : enumerate(I->operands())) {
     if (!canSplatOperand(I, OpIdx.index()))
```

Comment:
Instead of duplicating the existing check, is it possible to move this into canSplatOperand? I.e. check in there if `I` is a VPIntrinsic, and then also update the match on line 2873 to handle `m_Intrinsic<Intrinsic::experimental_vp_splat>`?